webapp/app/templates/components.html

```diff
@@ -71,27 +71,29 @@
 {% macro separated_field(field,first_field=False) -%}
     <fieldset id="{{ field.id }}">
         {{ field_label(field, class="text-input-field-label") }}
-        <div class="form-row-center">
+        <div class="form-row-center separated-field">
             {{ _field(field, first_field=first_field) }}
         </div>
         {{ field_errors(field) }}
     </fieldset>
 {%- endmacro %}
 
-{% macro field_label(field, class="") %}
+{% macro field_label(field, class="", optional=False) %}
     {%- set label = field.label -%}
     {%- set help = field.render_kw['help'] -%}
     {%- set detail = field.render_kw['detail'] -%}
 
     {% if field.type == 'RadioField' or field.type == 'UnlockCodeField' or field.type == 'IdNrField' or field.type == 'SteuerlotseDateField' %}
         {% if label.text %} {# Only show legend if there is a legend text to show #}
             <legend class="field-label {{ class }}">{{ label.text }}
+                {% if optional %}<span class="optional-hint">{{ _('form.optional') }}</span>{% endif %}
```

Comment:
Have you considered using `field.flags.required` to determine whether a field is required instead of passing the `optional` argument?

Reply:
This way, we'd display the "optional" text for every optional field. As far as I see it, this is not intended in the design. For example, for "Grad der Behinderung" the text is not expected to appear. So I figured to better set it explicitly, I could however set it directly in the field declaration by using render_kw of the field (and access is here via `field.render_kw["is_optional"]` as with the example input). I just thought we might want to avoid even more render_kw. What do you think?

Reply:
Oh, I see. This is clearly more complicated than I thought… I would need to understand to understand the design logic better to make a good call on this, so let's go with your solution!